### PR TITLE
Added model, controller and view for suggested councillors. 

### PR DESCRIPTION
**Pull request by hisayohorie**
It passes test for now!  some polishing may needed.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,6 +9,7 @@ tmp/sass-cache
 tmp/binlog
 tmp/rspec_guard_result
 tmp/capybara
+tmp/letter_opener
```
Comment (equivalentideas) [inline .gitignore]: What's this for?

File: app/assets/javascripts/suggested_councillors.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (equivalentideas) [inline app/assets/javascripts/suggested_councillors.coffee]: Let's remove this file, we can add a script file when we need it.

File: app/assets/stylesheets/suggested_councillors.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the suggested_councillors controller here.
```
Comment (equivalentideas) [inline app/assets/stylesheets/suggested_councillors.scss]: Let's remove this file, we can add a stylesheet when we need it.

File: app/helpers/suggested_councillors_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module SuggestedCouncillorsHelper
```
Comment (equivalentideas) [inline app/helpers/suggested_councillors_helper.rb]: I think we remove this too, and add a helper file when we need it.

File: config/routes.rb [hunk 5]
```diff
@@ -102,11 +102,14 @@ def matches?(request)
         get :per_week
       end
     end
+    resources :suggested_councillors, only:[:new, :create] 
     collection do
       get :test_feed
     end
   end
 
+  resources :councillors, only: [:new, :create]
```
Comment (equivalentideas) [inline config/routes.rb]: What's this for?

File: spec/controllers/suggested_councillors_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (equivalentideas) [inline spec/controllers/suggested_councillors_controller_spec.rb]: This call to a `rails_helper` file that we don't have is breaking the tests suit. This is another file created by the rails generator. I think just remove the whole file until we want to write controller tests for this.

File: spec/features/admin_loads_councillors_spec.rb [hunk 7]
```diff
@@ -59,7 +59,7 @@
 
   context "when some councillors have been removed from office" do
     around do |test|
-      Timecop.freeze(2016, 10, 11) { test.run }
+      Timecop.freeze(2016, 10, 15) { test.run }
```
Comment (equivalentideas) [inline spec/features/admin_loads_councillors_spec.rb]: This doesn't seem relevant to these changes? I think we added this as a test while we were pairing, but @henare has fixed the timezone issue with #1171 . We should remove this change from the branch.

File: spec/features/admin_loads_councillors_spec.rb [hunk 8]
```diff
@@ -72,7 +72,6 @@
       visit admin_authority_path(authority)
 
       click_button "Load Councillors"
-
```
Comment (equivalentideas) [inline spec/features/admin_loads_councillors_spec.rb]: I think this blank line between the test 'exercise' and 'verify' stages of the test makes it clearer. Would you mind adding it back @hisayohorie ?

File: spec/helpers/suggested_councillors_helper_spec.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
```
Comment (equivalentideas) [inline spec/helpers/suggested_councillors_helper_spec.rb]: Because we're removing the helper file, we should remove this test file as well. We try not to have pending test files sitting around, they never seem to get filled in.

File: spec/models/suggested_councillor_spec.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe SuggestedCouncillor, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (equivalentideas) [inline spec/models/suggested_councillor_spec.rb]: I think you could have some simple tests here for the validations. They seem quite important and worth having coverage for.

File: .gitignore (hunk 1 above)
Comment (hisayohorie) [inline .gitignore]: I was using the letter opener gem for other issues, indeed this is irrelevant to keep it here

File: config/routes.rb (hunk 5 above)
Comment (hisayohorie) [inline config/routes.rb]: this is the time when I was trying to add a column in councillor table. not relevant now.

File: spec/features/admin_loads_councillors_spec.rb (hunk 7 above)
Comment (hisayohorie) [inline spec/features/admin_loads_councillors_spec.rb]: Ok will do!

File: spec/features/admin_loads_councillors_spec.rb (hunk 8 above)
Comment (hisayohorie) [inline spec/features/admin_loads_councillors_spec.rb]: Sure I will!

File: app/controllers/suggested_councillors_controller.rb [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    @suggested_councillor = SuggestedCouncillor.new
+    @authority = Authority.find(params[:authority_id])
+  end
+
+  def create
+    @suggested_councillor = SuggestedCouncillor.new(suggested_councillor_params)
+    @authority = Authority.find(params[:authority_id])
+    @suggested_councillor.authority= @authority
+
+      if @suggested_councillor.save
```
Comment (equivalentideas) [inline app/controllers/suggested_councillors_controller.rb]: Is the indentation of this block off?

File: app/controllers/suggested_councillors_controller.rb [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+        flash[:notice] = "Thank you"
+        redirect_to authority_url(@authority.short_name_encoded)
```
Comment (equivalentideas) [inline app/controllers/suggested_councillors_controller.rb]: I'm interested in the choice to redirect to the authority page here. What's your thinking @hisayohorie ?

This is an area where having a single commit for this choice, with a description of why it was made, would make understanding the change easier.

File: app/controllers/suggested_councillors_controller.rb [hunk 13]
```diff
@@ -0,0 +1,25 @@
+class SuggestedCouncillorsController < ApplicationController
+  def new
+    @suggested_councillor = SuggestedCouncillor.new
+    @authority = Authority.find(params[:authority_id])
+  end
+
+  def create
+    @suggested_councillor = SuggestedCouncillor.new(suggested_councillor_params)
+    @authority = Authority.find(params[:authority_id])
+    @suggested_councillor.authority= @authority
```
Comment (equivalentideas) [inline app/controllers/suggested_councillors_controller.rb]: Would it be possible to assign this when the SuggestedCouncillor is initialized if `@authority` is assigned first?

File: spec/models/suggested_councillor_spec.rb [hunk 14]
```diff
@@ -0,0 +1,21 @@
+require 'spec_helper'
+
+RSpec.describe SuggestedCouncillor, type: :model do
+  it "should require a name" do
```
Comment (equivalentideas) [inline spec/models/suggested_councillor_spec.rb]: You might be able to make these tighter using `subject` (see http://www.betterspecs.org/#subject ) and `be_valid` (using [Rails method `.valid?`](http://api.rubyonrails.org/classes/ActiveRecord/Validations.html#method-i-valid-3F) with [rspec predicate matchers](https://relishapp.com/rspec/rspec-expectations/v/3-6/docs/built-in-matchers/predicate-matchers))

Do you think something like this would be a bit clearer + more succinct?

```
describe "is invalid without an email" do
  subject { SuggestedCouncillor.new(name: "Jane", email: nil, authority_id: 1) }
  it { is_expected.to_not be_valid }
end

describe "is invalid without a name" do
  subject { SuggestedCouncillor.new(name: nil, email: "example@email.org", authority_id: 1) }
  it { is_expected.to_not be_valid }
end

describe "is invalid with a blank name" do
  subject { SuggestedCouncillor.new(name: "", email: "example@email.org", authority_id: 1) }
  it { is_expected.to_not be_valid }
end
````

Or maybe it's time to start using a library for these common validation tests we have https://github.com/thoughtbot/shoulda-matchers#activemodel-matchers

File: app/controllers/suggested_councillors_controller.rb (hunk 13 above)
Comment (hisayohorie) [inline app/controllers/suggested_councillors_controller.rb]: @equivalentideas you are suggesting to use `@authority.suggested_councillors.build(suggested_councillor_params)` , am I right?

File: app/controllers/suggested_councillors_controller.rb (hunk 11 above)
Comment (hisayohorie) [inline app/controllers/suggested_councillors_controller.rb]: yes 😞

File: app/controllers/suggested_councillors_controller.rb (hunk 12 above)
Comment (hisayohorie) [inline app/controllers/suggested_councillors_controller.rb]: good question, because I don't exactly know where it should be redirected 😅 
i wasn't sure if i should make a 'Thank you' page. I thought one I think we know and the contributor know for sure is which authority the suggested councillor belongs to, so I redirected to authority page. But it is a very loose thinking and I would love to know your suggestion!

File: spec/models/suggested_councillor_spec.rb [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+    subject { SuggestedCouncillor.new(name: "Milla", email: "", authority_id: 1) }
+    it { is_expected.to_not be_valid }
+  end
+
+  describe "is invalid without an name" do
+    subject { SuggestedCouncillor.new(name: nil, email: "test@test.com", authority_id: 1) }
+    it { is_expected.to_not be_valid }
+  end
+
+  describe "is invalid without a blank name" do
```
Comment (equivalentideas) [inline spec/models/suggested_councillor_spec.rb]: Should that be "with a blank name" ?

File: app/controllers/suggested_councillors_controller.rb [hunk 16]
```diff
@@ -0,0 +1,24 @@
+class SuggestedCouncillorsController < ApplicationController
+  def new
+    @suggested_councillor = SuggestedCouncillor.new
+    @authority = Authority.find(params[:authority_id])
+  end
+
+  def create
+    @authority = Authority.find(params[:authority_id])
+    @suggested_councillor = @authority.suggested_councillors.build(suggested_councillor_params)
```
Comment (equivalentideas) [inline app/controllers/suggested_councillors_controller.rb]: Very nice 👌
